bin/update-bundler

```diff
@@ -0,0 +1,78 @@
+#!/bin/sh
+
+# Use Docker to update Bundler in several projects and open a PR for each one.
+#
+# Usage:
+#   update-bundler <BUNDLER_VERSION>
+# Example:
+#   update-bundler 2.5.4
+#
+# This script will update Bundler to the specified version in all projects listed in ~/.config/bundler-projects.
+#
+# You will require Docker and GitHub CLI to run this script.
+
+set -e
+
+BUNDLER_VERSION="$1"
+if [ -z "$BUNDLER_VERSION" ]; then
+    echo "Usage: Please provide a Bundler version. For example:"
+    echo "update-bundler 2.5.4"
+    exit 1
+fi
+
+BUNDLER_RELEASE="https://github.com/rubygems/rubygems/releases/tag/bundler-v$BUNDLER_VERSION"
+
+run_in_container() {
+	container_id=$1
+	shift
+	cmd=$*
+
+	docker exec --workdir /app "$container_id" bash -c "$cmd"
+}
+
+while IFS= read -r folder
+do
+	cd "$(eval echo $folder)"
```

Comment:
What's `$folder`? I'm curious about the `eval echo`.

Reply:
`$folder` is the directory that is currently being worked on from the list defined in `~/.config/bundler-projects`.

I use symbolic links for some of the paths the scripts needs to navigate to. Without `eval`, `echo $folder` would not recognise these symbolic links as directory paths.

Would renaming `$folder` to `$working_directory` (or something like this) make this clearer?

Reply:
Maybe readlink(1) is a more clear way to solve this (without introducing arbitrary evals).

```suggestion
	cd "$(readlink -f $folder)"
```

Reply:
Sounds good. I'll try this locally with `readlink -f` and see how it goes.

TIL!

Reply:
I am having challenges getting this to work locally using `readlink`. I think I may have my quotes or `$` setup incorrectly.

Would we want to replace `eval` at the end of the script with `readlink` as well?

Reply:
If `readlink` doesn't work then I guess `eval` is the right answer. That's too bad, but shell programming is like that.

Reply:
After learning more about `eval`, I hope I can get to a version of the script that works without it.